<launch>
    <arg name="first_robot1" default="robot1"/>
    <arg name="second_robot2" default="robot2"/>
    <!-- 地图名 || Map name -->
    <arg name="map" default="my_map"/>
    <!-- 加载地图 || Load map -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find yahboomcar_multi)/maps/$(arg map).yaml"/>
    <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find yahboomcar_multi)/rviz/tf_multi.rviz"/>

    <!-- ############################# first_robot1 ############################# -->

    <node pkg="yahboomcar_multi" type="broad_round.py" name="RoundBroad" output="screen" args="$(arg first_robot1)"/>
    <include file="$(find yahboomcar_multi)/launch/library/move_base_multi.launch">
        <arg name="ns" value="$(arg first_robot1)"/>
    </include>

    <!-- ############################# second_robot2 ############################# -->

    <node pkg="yahboomcar_multi" type="listener.py" name="RobotListener" output="screen"
          args="$(arg second_robot2) point1" ns="$(arg second_robot2)/">
        <rosparam param="linPIDparam">[1.0, 0, 1.0]</rosparam>
        <rosparam param="angPIDparam">[0.8, 0, 1.0]</rosparam>
    </node>
    <include file="$(find yahboomcar_multi)/launch/library/move_base_multi.launch">
        <arg name="ns" value="$(arg second_robot2)"/>
    </include>
</launch>
